### PR TITLE
papis: init at 0.5.2 + dependencies

### PR DESCRIPTION
**Pull request by nico202**
###### Motivation for this change
Papis is a powerful and highly extensible command-line based document and bibliography manager.
https://github.com/alejandrogallo/papis

###### Things done
Added all papis missing dependencies (arxiv2bib, habanero, papis-python-rofi, pylibgen)
Updated python-magic from 0.4.10 to 0.4.13

(HELP WANTED: I don't know how to check if all packages depending on python-magic still works, how do I check it?
I checked beancount and it installs, relatorio and tests are ok, s3cmd installs (has no tests). How do I ask nix to do this automatically for all the dependencies?)

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/office/papis/default.nix
Comment (dotlambda) [inline pkgs/applications/office/papis/default.nix]: Please specify `pname` and leave out `name`.
Same for the other packages.

File: pkgs/applications/office/papis/default.nix
Comment (dotlambda) [inline pkgs/applications/office/papis/default.nix]: `buildPythonApplication`

File: pkgs/applications/office/papis/default.nix
Comment (dotlambda) [inline pkgs/applications/office/papis/default.nix]: Please pass in `vim` directly.

File: pkgs/development/python-modules/papis-python-rofi/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/papis-python-rofi/default.nix]: This should be called python-rofi.

File: pkgs/top-level/python-packages.nix
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: python-rofi

File: pkgs/top-level/python-packages.nix
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Please move to `all-packages.nix` and use `python3Packages.callPackage`

File: pkgs/top-level/python-packages.nix
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix]: Would you mind moving this to `development/python-modules/`, specifying `pname`, `version` and using `fetchPypi`?

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix
Comment (dotlambda) [reply pkgs/top-level/python-packages.nix]: Ah, I see. papis-python-rofi seems to be a fork of python-rofi. Then it's all right like that,

**Comment by dotlambda** [comment]
For checking whether the dependents of python-magic still compile, use `nix-shell -p nox --run "nox-review pr 34761"`

**Review by dotlambda — COMMENTED** [review]
The papis package should be put in `pkgs/tools/...` since it's command line only.

**Comment by nico202** [comment]
1. [X] use pname instead of name
2. [X] move python-magic to its own file
3. [X] use vim instead of pkgs.vim
4. [X] move papis to all-packages, tools, use pythonApplication

**Comment by nico202** [comment]
nox is failing, but even the following command fails on unstable
```
nix-build -A python27Packages.graphite_pager -A fava -A python36Packages.python_magic -A diffoscope -A python27Packages.thumbor -A python27Packages.alot -A python27Packages.python_magic -A alot -A beancount -A python36Packages.relatorio -A s3cmd```

do we need to ask maintainers to fix the packages?

File: pkgs/development/python-modules/pylibgen/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pylibgen/default.nix]: So, Python 2 isn't supported but PyPy is?

File: pkgs/development/python-modules/pyparser/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyparser/default.nix]: `prePatch`
I think you should rather replace `==` by `>=` to make this patch future-proof.

File: pkgs/development/python-modules/pyparser/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyparser/default.nix]: Why not `fetchPypi`? `fetchPypi` is generally preferred because this makes it easier to override the `src`. However, in case e.g. tests are not uploaded to PyPI, it is better to use other sources.
The same applies for all other packages.

File: pkgs/development/python-modules/arxiv2bib/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/arxiv2bib/default.nix]: You could put a `with lib;` here to get rid of the `lib.` prefix if you want.

File: pkgs/development/python-modules/pyparser/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/pyparser/default.nix]: Wrong descripion and homepage.

**Comment by dotlambda** [comment]
If these packages do have maintainers, then yes it would nice to tell them about it.
But as long as all packages that were not broken beforehand compile fine after the update, you should not worry too much.

File: pkgs/development/python-modules/pyparser/default.nix
Comment (nico202) [reply pkgs/development/python-modules/pyparser/default.nix]: Thanks

File: pkgs/development/python-modules/pyparser/default.nix
Comment (nico202) [reply pkgs/development/python-modules/pyparser/default.nix]: yeah that was stupid by me, fixing

File: pkgs/development/python-modules/pylibgen/default.nix
Comment (nico202) [reply pkgs/development/python-modules/pylibgen/default.nix]: my fault

File: pkgs/development/python-modules/pyparser/default.nix
Comment (nico202) [reply pkgs/development/python-modules/pyparser/default.nix]: for this and for papis tests are missing, for one other they are failing. Fixed others

File: pkgs/development/python-modules/pyparser/default.nix
Comment (nico202) [reply pkgs/development/python-modules/pyparser/default.nix]: I'm not on bitbucket and I'm not willed to sign up, and I could not find  his mail address. He is @grimmy on github

File: pkgs/development/python-modules/pyparser/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/pyparser/default.nix]: postPatch

File: pkgs/development/python-modules/python-magic/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/python-magic/default.nix]: postPatch

File: pkgs/tools/misc/papis/default.nix
Comment (FRidh) [inline pkgs/tools/misc/papis/default.nix]: postPatch

File: pkgs/tools/misc/papis/default.nix
Comment (FRidh) [inline pkgs/tools/misc/papis/default.nix]: Why is this needed? Leave an explanation.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/papis/default.nix
Comment (nico202) [reply pkgs/tools/misc/papis/default.nix]: Done!

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.python-magic python3.pkgs.python-magic
@GrahamcOfBorg build python2.pkgs.arxiv2bib python3.pkgs.arxiv2bib
@GrahamcOfBorg build python2.pkgs.habanero python3.pkgs.habanero
@GrahamcOfBorg build python2.pkgs.papis-python-rofi python3.pkgs.papis-python-rofi
@GrahamcOfBorg build python3.pkgs.pylibgen
@GrahamcOfBorg build python2.pkgs.pyparser python3.pkgs.pyparser
@GrahamcOfBorg build papis

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.python_magic python3.pkgs.python_magic

**Comment by FRidh** [comment]
A commit is needed per logical change, which in this case is typically a commit per expression.

**Comment by nico202** [comment]
Should python_magic be renamed to python-magic? Any guideline on the naming convention? I don't like the underscore but I kept it as it was

**Comment by dotlambda** [comment]
> Ran 0 tests in 0.000s

Please specify the appropriate `checkPhase` for arxiv2bib
For papis-python-rofi, you'll need to set `doCheck = false` and leave a comment that there are no tests

**Comment by dotlambda** [comment]
pylibgen also needs an appropriate `checkPhase`

**Comment by nico202** [comment]
Fixed tests, let me know

File: pkgs/development/python-modules/arxiv2bib/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/arxiv2bib/default.nix]: `checkInputs`

File: pkgs/development/python-modules/habanero/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/habanero/default.nix]: `pname`

File: pkgs/development/python-modules/python-magic/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/python-magic/default.nix]: I don't think this is necessary. Is it, @FRidh?

Comment (FRidh) [reply pkgs/development/python-modules/python-magic/default.nix]: No, it is not. The patch introduces it so it can be found during run-time. I doubt it needs to be explicitly added during build-time.

**Comment by nico202** [comment]
I had to change dateparser too because it was needed in order to run habanero tests. However tests on python3 are failing (as per comment). Failing tests are only 3 and because of this error:

AssertionError: False is not true : Didn't found any of the expected messages (['year is out of range', "('year must be in 1..9999'"]) -- message was: ValueError('year -2986 is out of range',)

So it might be possible to fix it upstream, I'll bump the issue

**Comment by dotlambda** [comment]
I just created a PR for dateparser :D
We might want to merge #34768 first. Then you can rebase.

**Comment by nico202** [comment]
:D no problem, I'll wait for dateparser merge.
Thanks

**Comment by dotlambda** [comment]
@nico202 You can remove your commit regarding dateparser and rebase on master.

**Comment by nico202** [comment]
@dotlambda thanks for everything, I know it's been a long review I'm sorry, I hope this time it's ok

File: pkgs/development/python-modules/arxiv2bib/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/arxiv2bib/default.nix]: `${python.interpreter}` instead of `python`

File: pkgs/development/python-modules/pylibgen/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/pylibgen/default.nix]: `${python.interpreter}` instead of `python`

**Review by FRidh — CHANGES_REQUESTED** [review]
Almost there ;)

File: pkgs/development/python-modules/arxiv2bib/default.nix
Comment (nico202) [reply pkgs/development/python-modules/arxiv2bib/default.nix]: ehm python is undefined

Comment (nico202) [reply pkgs/development/python-modules/arxiv2bib/default.nix]: I add it to dependencies

**Comment by nico202** [comment]
Changed papis github url since yesterday it was moved to papis/papis

**Pull request merged**
